contrib/bsp/src/mill/contrib/bsp/MillBuildServer.scala

```diff
@@ -132,24 +132,28 @@ class MillBuildServer(evaluator: Evaluator, bspVersion: String, serverVersion: S
       val millBuildTargetId = getMillBuildTargetId(evaluator)
 
       val items = dependencySourcesParams.getTargets.asScala
-        .filter(_ != millBuildTargetId)
         .foldLeft(Seq.empty[DependencySourcesItem]) { (items, targetId) =>
-          val module = getModule(targetId, modules)
-          val sources = evaluateInformativeTask(
-            evaluator,
-            module.resolveDeps(T.task(module.compileIvyDeps() ++ module.transitiveIvyDeps()), sources = true),
-            Agg.empty[PathRef]
-          )
-          val unmanaged = evaluateInformativeTask(
-            evaluator,
-            module.unmanagedClasspath,
-            Agg.empty[PathRef]
-          )
+          val all = if (targetId == millBuildTargetId) {
+            Try(getClass.getClassLoader.asInstanceOf[SpecialClassLoader]).fold(
+              _ => Seq.empty,
+              _.allJars.filter(url => isSourceJar(url) && exists(Path(url.getFile))).map(_.toURI.toString)
+            )
```

Comment:
Now we get the source files for the build dependencies filtering with `isSourceJar`.